components/SignUp.js

```diff
@@ -23,29 +23,33 @@ const SignUp = ({ navigation }) => {
         username: "",
         password: "",
         confirmPassword: "",
+        email: "",
     });
     const errorMsg = useSelector((state) => state.auth.error);
     const dispatch = useDispatch();
     const usernameInput = useRef(null);
     const passwordInput = useRef(null);
     const confirmPasswordInput = useRef(null);
+    const emailInput = useRef(null);
 
     useEffect(() => {
         if (errorMsg != null) {
             emptyFieldsAlert();
         }
     }, [errorMsg]);
 
-    useEffect(() => {
-        if (credentials.password != credentials.confirmPassword) {
-            passwordMatch();
-        }
-    }, []);
+    function validateEmail(email) {
+        var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
Love this regex. I'd recommend making it a const, and putting it in a utility folder somewhere in case you have to use it somewhere else.